src/pages/Typing/store/index.ts

```diff
@@ -0,0 +1,230 @@
+import { Word } from '@/typings'
+import { createContext } from 'react'
+
+export type ChapterData = {
+  words: Word[]
+  index: number
+  correctCount: number
+  wrongCount: number
+  wrongWordIndexes: number[]
+}
+export type TimerData = {
+  startTime: Date
+  time: number
+  accuracy: number
+  wpm: number
+}
+
+export type TypingState = {
+  chapterData: ChapterData
+  timerData: TimerData
+  isTyping: boolean
+  isFinished: boolean
+  isShowSkip: boolean
+  isWordVisible: boolean
+}
```

Comment:
```suggestion
  hasError: boolean
}
```

Reply:
Add a new state `hasError` indicating whether the user has typed the current word incorrectly. You should prevent typing if `hasError` is true. Otherwise, you will get this if you smash the keyboard:

<img width="691" alt="image" src="https://user-images.githubusercontent.com/2239547/228589809-b8be115f-6fde-4c5c-8c5d-89eef659ec28.png">